source/assets/scripts/jsonInterpreter.js

```diff
@@ -1,143 +1,145 @@
-// calendar.js
-
-// Run the init() function when the page has loaded
-window.addEventListener('DOMContentLoaded', init);
-
-const max_month_days = 60;
-const test_month_days = 30;
-var events_list = [];
-
-var start_col = 0;
-var start_row = 0;
-var num_disp_days = test_month_days;
-
-// Starts the program, all function calls trace back here
-function init() {
-
-  // initialize events list
-  for(let i = 0; i < max_month_days; i++) {
-    events_list.push([]);
+/**
+ * JsonInterpreter.js
+ * @author Yuelin Dai
+ * @summary File contains JSON interpreter.
+ * 
+ * Created at : 2022-11-17 6:30 AM
+ * Last Modified : 2022-11-17 6:30 AM
+ */
+
+/**
+ * JSON string used for test
+ */
+const testJson = `
+{
+    "lastUpdated" : "",
+    "title" : "title1",
+    "calendarID" : "",
+
+    "usersList" : [
+        {
+            "firstName" : "john",
+            "lastName" : "doe",
+            "username":"admin",
+            "password":"pwd",
+            "profieldID" : "123456",
+            "calenarID" : ""
+        }
+    ],
+    "eventsList" : [
+        {
+          "startDay" : "11/01/22 11:11",
+          "endDay" : "11/02/22 11:11",
+          "eventName":"e1",
+          "users":[0], 
+          "location":"",
+          "description":""
+        },
+        {
+          "startDay" : "10/01/22 11:11",
+          "endDay" : "10/02/22 11:11",
+          "eventName":"e1",
+          "users":[0], 
+          "location":"",
+          "description":""
+        }
+    ],
+    "tasksList" : [
+        {
+            "taskName" : "task1",
+            "tags" : " ",
+            "dueDate" : "10/01/22 11:11",
+            "description" : " ",
+            "complete" : "false",
+            "users" : [0]
+        }
+    ]
   }
-  events_list[5].push('event 1');
-  events_list[8].push('event 1');
-  events_list[19].push('event 1');
-  console.log(events_list.length);
-  
-  refreshCalendar();
-
-  initFormHandler();
+`;
+
+/**
+ * Get the corresponding string of a day 
+ * @param ind - order number of the day (1 - 7)
```

Comment:
JSDoc comments should also have a {number} so we can type check the variable. In this case line 62 should be @param {number} ind - order number of the day (1 - 7)

Reply:
do this for all of @param